tt_metal/hw/inc/debug/sanitize_noc.h

```diff
@@ -214,8 +214,14 @@ void debug_sanitize_noc_and_worker_addr(
             ((uint64_t)NOC_CMD_BUF_READ_REG(noc_id, NCRISC_WR_CMD_BUF, NOC_RET_ADDR_MID) << 32) | \
             ((uint64_t)NOC_CMD_BUF_READ_REG(noc_id, NCRISC_WR_CMD_BUF, NOC_RET_ADDR_LO)),     \
         NOC_CMD_BUF_READ_REG(noc_id, NCRISC_WR_CMD_BUF, NOC_TARG_ADDR_LO),                    \
-        NOC_CMD_BUF_READ_REG(noc_id, NCRISC_WR_CMD_BUF, NOC_AT_LEN_BE));                      \
-    debug_insert_delay((uint8_t)TransactionWrite);
+        NOC_CMD_BUF_READ_REG(noc_id, NCRISC_WR_CMD_BUF, NOC_AT_LEN_BE));
+#define DEBUG_SANITIZE_NOC_ADDR_FROM_STATE(noc_id)                                                 \
+    DEBUG_SANITIZE_NOC_ADDR(                                                                       \
+        noc_id,                                                                                    \
+            ((uint64_t)NOC_CMD_BUF_READ_REG(noc_id, NCRISC_WR_REG_CMD_BUF, NOC_TARG_ADDR_COORDINATE) << NOC_ADDR_COORD_SHIFT) | \
```

Comment:
This is reading state from `NCRISC_WR_REG_CMD_BUF`, what's the difference between that one and `NCRISC_WR_CMD_BUF` that the previous macro uses? Also wondering if we need a RD and WR version here

Reply:
They're just for separating writes onto different cmd bufs. Specifically in `noc_nonblocking_api.h` we comment that `NCRISC_WR_CMD_BUF ` is for large writes, and `NCRISC_WR_REG_CMD_BUF ` is for small writes, and `NCRISC_WR_REG_CMD_BUF ` was what was being used in the original dataflow_api fn.

Reply:
I think I will uplift the cmd buf for this macro to be an arg instead of hardcoding.

The original `DEBUG_SANITIZE_NOC_ADDR` is used for this inline write as well as atomic inc, and atomic inc uses a different cmd buf `NCRISC_AT_CMD_BUF`. If we add stateful apis for atomics then this will be needed.